pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -356,6 +384,67 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	var typhaTLSSecret, nodeTLSSecret *corev1.Secret
+	typhaCAConfigMap := &corev1.ConfigMap{}
+	dpiList := &v3.DeepPacketInspectionList{}
+	if err := r.client.List(ctx, dpiList); err != nil {
+		r.status.SetDegraded("Failed to retrieve DeepPacketInspection resource", err.Error())
+		return reconcile.Result{}, err
+	}
+	if len(dpiList.Items) > 0 {
+		nodeTLSSecret, err = utils.GetSecret(ctx, r.client, render.NodeTLSSecretName, rmeta.OperatorNamespace())
+		if err != nil {
+			reqLogger.Error(err, fmt.Sprintf("Failed to retrieve %s secret", render.NodeTLSSecretName))
+			r.status.SetDegraded(fmt.Sprintf("Failed to retrieve %s secret", render.NodeTLSSecretName), err.Error())
+			return reconcile.Result{}, err
+		}
+
+		typhaTLSSecret, err = utils.GetSecret(ctx, r.client, render.TyphaTLSSecretName, rmeta.OperatorNamespace())
+		if err != nil {
+			reqLogger.Error(err, fmt.Sprintf("Failed to retrieve %s secret", render.TyphaTLSSecretName))
+			r.status.SetDegraded(fmt.Sprintf("Failed to retrieve %s secret", render.TyphaTLSSecretName), err.Error())
+			return reconcile.Result{}, err
+		}
+
+		// If TLS secrets are not available, degrade.
+		if nodeTLSSecret == nil || typhaTLSSecret == nil {
+			reqLogger.Error(err, fmt.Sprintf("Waiting for both %s and %s secrets to be available", render.NodeTLSSecretName, render.TyphaTLSSecretName))
+			r.status.SetDegraded(fmt.Sprintf("Waiting for both %s and %s secrets to be available", render.NodeTLSSecretName, render.TyphaTLSSecretName), "")
+			return reconcile.Result{}, err
+		}
+
+		err = r.client.Get(ctx, types.NamespacedName{Name: render.TyphaCAConfigMapName, Namespace: rmeta.OperatorNamespace()}, typhaCAConfigMap)
```

Comment:
Don't you need to check if the typha config map is nil?

It's also surprising that there isn't a GetConfigMap function, like the GetSecret function.

Reply:
If the configmap is not available client throws error here. I was expecting same behaviour from GetSecret, but it returns nil if secret is not available, only that needs a nil check.

Reply:
Oh right right, the GetSecret uses the client in this same way but swallows the not found error. Forget this comment.